src/error.rs

```diff
@@ -1,24 +1,60 @@
+use std::error::Error as StdError;
+use std::fmt;
 use std::str::Utf8Error;
 
 use quick_xml::Error as XmlError;
-use thiserror::Error;
 
-#[derive(Debug, Error)]
+#[derive(Debug)]
 /// An error that occurred while performing an Atom operation.
 pub enum Error {
     /// Unable to parse XML.
-    #[error("{0}")]
-    Xml(#[from] XmlError),
+    Xml(XmlError),
     /// Unable to parse UTF8 in to a string.
-    #[error("{0}")]
-    Utf8(#[from] Utf8Error),
+    Utf8(Utf8Error),
     /// Input did not begin with an opening feed tag.
-    #[error("input did not begin with an opening feed tag")]
     InvalidStartTag,
     /// Unexpected end of input.
-    #[error("unexpected end of input")]
     Eof,
     /// The format of the timestamp is wrong.
-    #[error("timestamps must be formatted by RFC3339, rather than {0}")]
     WrongDatetime(String),
 }
+
+impl StdError for Error {
+    fn cause(&self) -> Option<&dyn StdError> {
+        match *self {
+            Error::Xml(ref err) => Some(err),
+            Error::Utf8(ref err) => Some(err),
+            Error::InvalidStartTag => None,
+            Error::Eof => None,
+            Error::WrongDatetime(_) => None,
+        }
+    }
+}
+
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match *self {
+            Error::Xml(ref err) => err.fmt(f),
```

Comment:
`err.fmt(f)` is ambiguous (Both `Display` and `Debug` are implemented for `XmlError`).

Reply:
Thank you! I had based this off [the error implementation from rss](https://github.com/rust-syndication/rss/blob/004aebdba2ce4c60a949b05ff7c971c3730b3fdf/src/error.rs). Addressed both these in the PR.